### PR TITLE
auto pass http_proxy to container

### PR DESCRIPTION
**Pull request by jamescassell**
works toward fixing #765

### CODE REVIEW

File: run.go [hunk 1]
```diff
@@ -939,6 +939,13 @@ func (b *Builder) configureEnvironment(g *generate.Generator, options RunOptions
 		}
 	}
 
+	for _, envSpec := range []string{"http_proxy", "https_proxy", "no_proxy"} {
```
Comment (rhatdan) [inline run.go]: Need an option --http-proxy, which is defaulted to true to turn this off in buildah run and buildah bud.

buildah bud --http-proxy=false ...

Comment (rhatdan) [reply run.go]: Also I thought you wanted to test upper case.

**Comment by rhatdan** [comment]
bot, add author to whitelist

**Comment by rhatdan** [comment]
bot, test pull request

**Comment by rhatdan** [comment]
Would like a test on this.  Just to make sure the environment variables get passed.  
Please make sure the environment variables do not end up in the container image.

Once you add command line options, you need to add them to man pages and command completions.

File: run.go (hunk 1 above)
Comment (jamescassell) [reply run.go]: I've pushed an update that includes more proxy vars, including upper case.  The initial push was the subset vars that I've found that work in the most common cases.

Still needs an option, which I haven't figured out how to do yet.

**Comment by jamescassell** [comment]
> Would like a test on this. Just to make sure the environment variables get passed.
> Please make sure the environment variables do not end up in the container image.
> 

These env vars are evaluated each time a 'buildah run' command is executed.  If you unset some of these vars between 'buildah run' commands, they cease to be present during the subsequent 'bulidah run command'

I've also confirmed that these do not persist 'buildah commit', both by inspecting the metadata and by 'podman run --rm -it <image> env'

If you meant automated testing, I'm not sure where to start for that.

> Once you add command line options, you need to add them to man pages and command completions.

I might need a little help adding the command line options.  I'm sure I can figure out the man pages and bash completions.

**Comment by rhatdan** [comment]
Tests are failing, because your second PR was not signed.  It is better if you squash your commits together.

git rebase -i origin
This will load a screen which has the "f" or "fixup" option, which you can use to squash your PRs together.

**Comment by rhatdan** [comment]
As far as adding a flag, first thing you need to do is add a build options.

in buildah.go look for

CommonBuildOptions

Add you HttpProxy boolean option to this.

Now in your existing code wrap it with 
```
if b.CommonBuildOptions.HttpProxy {
}
```
 in cmd/buildah/from.go and cmd/buildah/from.go
Set the HttpProxy values in the section that sets up the BuilderOptions.

Now in pkg/cli/common.go look for FromAndBudResults, add a boolean there.  
Now look for how CPUQuota is handled and add similar calls for HttpProxy.

**Comment by jamescassell** [comment]
I've pushed my (failed) attempts to make the functionality optional.  The result complies, but the option is always disabled and the command line flags are not honored.  I'm likely missing something in connecting the setting of the option and reading of the option.

Side note: I also tested the original functionality of this PR in `buildah bud` and it worked exactly as expected.

File: pkg/cli/common.go [hunk 2]
```diff
@@ -54,6 +54,7 @@ type BudResults struct {
 	DisableContentTrust bool
 	File                []string
 	Format              string
+	HttpProxy    bool
```
Comment (rhatdan) [inline pkg/cli/common.go]: You need to run your code changes through `gofmt -w -s`, to be able to pass tests.

File: cmd/buildah/bud.go [hunk 3]
```diff
@@ -272,7 +272,6 @@ func budCmd(c *cobra.Command, inputArgs []string, iopts budResults) error {
 		ForceRmIntermediateCtrs: iopts.ForceRm,
 		BlobDirectory:           iopts.BlobCache,
 		Target:                  iopts.Target,
-		HttpProxy:               iopts.HttpProxy,
```
Comment (rhatdan) [inline cmd/buildah/bud.go]: Why are you reverting these.  I believe the original was correct.

**Comment by jamescassell** [comment]
I've gotten the '--http-proxy' option to work for 'buildah run' but not for 'buildah from' or 'buildah bud'.

Likely, I just haven't hooked things up properly.  Seems like there are several layers of indirection going on to get those hooked up.

I'll rebase everything in the end once I've gotten the functionality working; such rebase will ensure code formatting and signed-off-by headers are up to standards.

**Comment by rhatdan** [comment]
@jamescassell You would only see the --http-proxy working when doing a buildah run after you used it in the buildah from.

**Comment by rhatdan** [comment]
@rh-atomic-bot Could you squash all of your commits together

git rebase -i origin

Then switch each commit to a "f" after the fist one.

**Comment by jamescassell** [comment]
ready_for_review

Also, is there an easy way to add this to podman to have similar functionality when running containers there?

File: pkg/cli/common.go (hunk 2 above)
Comment (jamescassell) [reply pkg/cli/common.go]: Fixed

File: docs/buildah-bud.md [hunk 4]
```diff
@@ -222,6 +222,14 @@ Recognized formats include *oci* (OCI image-spec v1.0, the default) and
 Note: You can also override the default format by setting the BUILDAH\_FORMAT
 environment variable.  `export BUILDAH_FORMAT=docker`
 
+**--http-proxy**
+
+By default proxy environment variables are passed into the container if set
```
Comment (rhatdan) [inline docs/buildah-bud.md]: Add `Defaults to true`

File: docs/buildah-from.md [hunk 5]
```diff
@@ -171,6 +171,14 @@ The [username[:password]] to use to authenticate with the registry if required.
 If one or both values are not supplied, a command line prompt will appear and the
 value can be entered.  The password is entered without echo.
 
+**--http-proxy**
+
+By default proxy environment variables are passed into the container if set
```
Comment (rhatdan) [inline docs/buildah-from.md]: Add `Defaults to true.`

**Comment by rhatdan** [comment]
The tests don't like the naming of HttpProxy
```
# Run gofmt on version 1.11 and higher
buildah.go:286:2:warning: struct field HttpProxy should be HTTPProxy (golint)
Makefile:66: recipe for target 'validate' failed
make: *** [validate] Error 1
```
Rename it to make golint happy.

Also add --http-proxy options to contrib/completions/bash/buildah

And squash your PRs into one PR.

**Comment by rhatdan** [comment]
@jamescassell We will vendor this code into podman and that will handle the podman build case.  But we will need work to get this into 
podman run and podman create.

**Comment by jamescassell** [comment]
Closing since a squash was requested.  See #1525

**Pull request closed without merging**